packages/discord.js-utilities/src/lib/PaginatedMessages/PaginatedMessage.ts

```diff
@@ -867,16 +868,14 @@ export class PaginatedMessage {
 				} else {
 					await this.response.reply(page as WebhookEditMessageOptions);
 				}
-			} else {
+			} else if (isMessageInstance(this.response)) {
 				await this.response.edit(page as WebhookEditMessageOptions);
 			}
 		} else if (runsOnInteraction(messageOrInteraction)) {
-			this.response = messageOrInteraction;
-
-			if (this.response.replied || this.response.deferred) {
-				await this.response.editReply(page as MessageOptions);
+			if (messageOrInteraction.replied || messageOrInteraction.deferred) {
+				this.response = await messageOrInteraction.editReply(page);
 			} else {
-				await this.response.reply(page as MessageOptions);
+				this.response = await messageOrInteraction.reply({ fetchReply: true, ...page });
```

Comment:
If we provide fetchReply to false in the lage it won't fetch, if we provide ephemeral in the page it won't work

Reply:
As far as TS is concerned it's not possible for a page to be `ephemeral`. `page` is of type `MessageOptions | WebhookEditMessageOptions`, which are https://github.com/discordjs/discord.js/blob/stable/typings/index.d.ts#L5144-L5155 and https://github.com/discordjs/discord.js/blob/stable/typings/index.d.ts#L5673-L5676

As for the `fetchReply`, that too is not possible

Anyway for sanity checks I made it `{ ...page, fetchReply: true, ephemeral: false }`. This does still work for pre-existing ephemeral responses because those would use `editReply` (i.e. if I first defer then call `.run`)